src/controllers/api/friends.js

```diff
@@ -0,0 +1,64 @@
+const { User } = require("../../models");
+
+const addFriend = async (req, res) => {
+  try {
+    const { id } = req.body;
+    const { userId } = req.params;
+
+    const user = await User.findById(userId);
+
+    if (id) {
```

Comment:
This is perfectly fine however I have tended to see that it is usually the `unhappy path` condition that we add to an `if` block to filter that out from the normal flow of the function. Given that the `try` block's main purpose is to send the good response, we probably want to see that in the main flow of the code. So I would refactor this (and any other controllers) to look like:

```
if (!id) {
return res.status(400).json({
      success: false,
      error: "Please enter the user ID for the friend.",
    });
 }

 const friend = await User.findByIdAndUpdate(
        userId,
        {
          $push: { friends: id },
        },
        { returnDocument: "after" }
      );

      return res.status(201).json({
        success: true,
        data: friend,
      });


```